python/dask_cudf/dask_cudf/groupby.py

```diff
@@ -83,13 +83,26 @@ def __getitem__(self, key):
         g._meta = g._meta[key]
         return g
 
+    @_dask_cudf_nvtx_annotate
+    def _columns_not_in_by(self):
+        """Generator of the columns contained in the groupby agg result"""
+
+        if isinstance(self.by, list):
+            for c in self.obj.columns:
+                if c not in self.by:
+                    yield c
+        else:
+            for c in self.obj.columns:
+                if c != self.by:
+                    yield c
+
     @_dask_cudf_nvtx_annotate
     @_check_groupby_supported
     def count(self, split_every=None, split_out=1):
         return groupby_agg(
```

Comment:
It looks like these aggregations are still highly repetitive. Could we do this and reduce a few dozen lines?

```python
def _make_groupby_agg(self, agg_name, split_every=None, split_out=1):
    return groupby_agg(
        self.obj,
        self.by,
        {c: agg_name for c in self._columns_not_in_by()},
        split_every=split_every,
        split_out=split_out,
        sep=self.sep,
        sort=self.sort,
        as_index=self.as_index,
        **self.dropna,
    )
```
and then call it in each function?

```python
@_dask_cudf_nvtx_annotate
@_check_groupby_supported
def count(self, split_every=None, split_out=1):
    return self._make_groupby_agg("count", split_every, split_out)
```

Reply:
This suggestion may also apply to `Series` groupby, so I'd do this in a later PR and keep the scope constrained here.

Reply:
That makes sense to me - happy to open up a separate PR handling this 🙂